src/map.h

```diff
@@ -729,10 +730,14 @@ class map
         // TODO: fix point types (remove the first overload)
         std::vector<tripoint> route( const tripoint &f, const tripoint &t,
                                      const pathfinding_settings &settings,
-        const std::unordered_set<tripoint> &pre_closed = {{ }} ) const;
+        std::function<bool( const tripoint & )> avoid = []( const tripoint & ) {
```

Comment:
If you pass the function as `const std::function<...> &avoid` instead you can avoid copies, also you can pass them directly without the lambda wrappers like in monmove.cpp line 1032, pathfinding.cpp 215 etc

Reply:
i think the copy is quite cheap here generally (should just be ~16 bytes most of the time so can happen on the stack even), but we can elide it so we might as well :)